compose/config/validation.py

```diff
@@ -216,15 +234,17 @@ def _parse_valid_types_from_schema(schema):
 
 def validate_against_fields_schema(config):
     schema_filename = "fields_schema.json"
-    return _validate_against_schema(config, schema_filename)
+    format_checkers = ["ports", "environment"]
+    return _validate_against_schema(config, schema_filename, format_checkers)
 
 
 def validate_against_service_schema(config, service_name):
     schema_filename = "service_schema.json"
-    return _validate_against_schema(config, schema_filename, service_name)
+    format_checkers = ["ports"]
```

Comment:
I feel like this is outside scope of this PR. Can we come back to that later in a separate PR as I think depending on how some of the future extend/import work is going to happen, this indeed will need to be re-factored but I'd like space for us to have a separate discussion around that.

Heh, yeh abstract base services are a bit of a pain.